server/uno_test.go

```diff
@@ -179,3 +179,87 @@ func TestDrawCard(t *testing.T) {
 // 	allCards[players[0]] = make([]model.Card, 0)
 // 	assert.Equal(t, "player1", checkForWinner())
 // }
+
+
+func TestDealCards(t *testing.T) {
+
+	// Test passing in a bogus game id, we should get an error
+	game, err := dealCards("Bogus game id")
+
+	// Assert that we got an actual error
+	assert.NotNil(t, err, "We did not error on a bogus game id")
+
+	// Generate real game in database and real player
+	database, _ := db.GetDb()
+	game, player := setupGameWithPlayer(database)
+
+	// Test Drawing a card with a full deck and real player
+	game, err = dealCards(game.ID)
+	player = &game.Players[game.CurrentPlayer] //getting from the game who the current player is
+
+	// Assert that no error occured, the player has a new card and the draw pile
+	// has one less card
+	assert.Nil(t, err, "Failed to deal cards.")
+	assert.Equal(t, 7, len(player.Cards))
+	assert.Equal(t, 101, len(game.DrawPile))
+	assert.Equal(t, 1, len(game.DiscardPile))
+	
+	// Create additional players and add them to the game
+	player2, _ := database.CreatePlayer("Player 2")
+	player3, _ := database.CreatePlayer("Player 3")
+	player4, _ := database.CreatePlayer("Player 4")
+	player5, _ := database.CreatePlayer("Player 5")
+
+	game, _ = database.JoinGame(game.ID, player2.ID)
+	//Have to save in between each player being added or the game state wont recall any but the last
+	database.SaveGame(*game)
+	game, _ = database.JoinGame(game.ID, player3.ID)
+	database.SaveGame(*game)
+	game, _ = database.JoinGame(game.ID, player4.ID)
+	database.SaveGame(*game)
+	game, _ = database.JoinGame(game.ID, player5.ID)
+	database.SaveGame(*game)
+
+
+	//refresh the drawPile and the discardPile
+	game.DrawPile = generateShuffledDeck()
+	game.DiscardPile = []model.Card{}
+
+	// Test Drawing a card with a full deck and multiple players
+	game, err = dealCards(game.ID)
+	// Assert that no error occured, the player has a new card and the draw pile
+	// has one less card
+	assert.Nil(t, err, "Failed to deal multiple players cards.")
+	for _, player := range game.Players {
+		assert.Equal(t, 7, len(player.Cards))
+		}
+	assert.Equal(t, 73, len(game.DrawPile))
+	assert.Equal(t, 1, len(game.DiscardPile))
+	
+	//refresh the drawPile and the discardPile
+	game.DrawPile = generateShuffledDeck()
+	game.DiscardPile = []model.Card{}
+	//remove all but the top 5 cards so we will run out for sure part way through dealing
+	game.DrawPile = game.DrawPile[:5]
+	database.SaveGame(*game)
+
+	//run deal cards
+	game, err = dealCards(game.ID)
+	//check if the draw pile refreshed
+	assert.Equal(t, 78, len(game.DrawPile))
+
+
+	//database.SaveGame(*game)		//Use this if you make a modification to the game, drawcard will save it for you but need to call whenever you change game state
+}
+
+
+
+
+
```

Comment:
Can we reduce the amount of whitespace here?

Reply:
done